tests/backup/backup_nfs_test.go

```diff
@@ -203,3 +204,211 @@ var _ = Describe("{DeleteNfsExecutorPodWhileBackupAndRestoreInProgress}", func()
 		CleanupCloudSettingsAndClusters(backupLocationMap, "", "", ctx)
 	})
 })
+
+// CloudSnapshotMissingValidationForNFSLocation this TC validates the Cloud Snapshot missing status for backups after deleting the Cloud Snapshots on the NFS location
+var _ = Describe("{CloudSnapshotMissingValidationForNFSLocation}", func() {
+	var (
+		bkpLocationName            string
+		backupLocationUID          string
+		customResourceBackupName   string
+		clusterUid                 string
+		appNamespaces              []string
+		appContexts                []*scheduler.Context
+		scheduledAppContexts       []*scheduler.Context
+		globalBucket               string
+		backupNames                []string
+		backupLocationMap          = make(map[string]string)
+		periodicSchedulePolicyName string
+		singleNSScheduleName       string
+		singleNSScheduledBackup    string
+		multipleNSScheduleName     string
+		multipleNSScheduledBackup  string
+		scheduleNames              []string
+		periodicSchedulePolicyUid  string
+	)
+
+	JustBeforeEach(func() {
+		StartPxBackupTorpedoTest("CloudSnapshotMissingValidationForNFSLocation", "This TC validates the Cloud Snapshot missing status for backups after deleting the Cloud Snapshots on the NFS location", nil, 86094, Sabrarhussaini, Q1FY24)
+		log.InfoD("Scheduling Applications")
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				appNamespaces = append(appNamespaces, namespace)
+				scheduledAppContexts = append(scheduledAppContexts, ctx)
+			}
+		}
+		log.Infof("The list of namespaces deployed are", appNamespaces)
+	})
+
+	It("To validate the backup status when the cloud snapshots are deleted", func() {
+		Step("Validate applications", func() {
+			log.InfoD("Validating applications")
+			ValidateApplications(scheduledAppContexts)
+		})
+
+		Step("Creating NFS backup location", func() {
+			log.InfoD("Creating NFS backup location")
+			backupLocationProviders := getProviders()
+			for _, provider := range backupLocationProviders {
+				globalBucket = getGlobalBucketName(provider)
+				bkpLocationName = fmt.Sprintf("%s-%s-%s", provider, globalBucket, RandomString(10))
+				backupLocationUID = uuid.New()
```

Comment:
globalBucket is not used except in bkpLocationName 
getGlobalBucketName(provider) is called again for CreateBackupLocation
We can use 'globalBucket' only in CreateBackupLocation call instead of using getGlobalBucketName(provider) again

Reply:
Also this is specific to NFS. We should not iterate over backupLocationProviders  as if we run this on aws pipeline..it should log error saying :Not supported on this location provider
At present , backup location will not be created and it will fail. User might get confused as why is it failing

Reply:
Addressed